src/links.ts

```diff
@@ -63,6 +63,22 @@ export function getLinks(source: string, baseUrl: string): ParsedUrl[] {
   const sanitized = links
     .filter(link => !!link)
     .map(link => parseLink(link, realBaseUrl));
+  // Filter out the links with the same url.href like
+  // `google.cloud.vision.v1.IAddProductToProductSetRequest#.decode` and `google.cloud.vision.v1.IAddProductToProductSetRequest#.encode`
+  const uniqueHref = new Set();
+  const uniqueLinks: ParsedUrl[] = [];
+  sanitized.forEach(link => {
```

Comment:
I don't quite follow how this code works.  It looks like we're creating lists of unique links and unique hrefs, but those have no effect on the `sanitized` collection as far as I can tell.

Reply:
Sorry just made some changes to the code, `uniqueLinks` should be returned. 
For 
Link1: /page (url: `page.html`) 
Link2: /page#fragment1 (url: `page.html`) 
Link3: /page#fragment2 (url: `page.html`) here it wants to make sure only one link (/page) is added to the queue instead of three. Then it will return set of paredLinks with unique URL.